### PR TITLE
New importer package

### PR DESCRIPTION
**Pull request by gimenete**
This PR adds a new package. Maybe there are a few tweaks to do it but I think it's in a good shape to be used. More information on how to use it in the README and in the command's `--help`.

The main goal in this PR was to make a CLI app. I've tried to make the core functionality abstracted enough so it can be used as an API, but I haven't tried that yet. I'll do another PR for further changes regarding that.

Right now the only format supported is CSV, but similarely I've tried to abstract things enough to support more file formats in the future.

This PR supports creating a table if it doesn't exist as well as updating an existing one. It also supports creating new columns in an existing columns if neccessary. It tries to guess the right data type for each column, etc.

### CODE REVIEW

File: .changeset/config.json
Comment (SferaDev) [inline .changeset/config.json:5]: Not sure if we want to link all dependency versions. The codegen kind of made sense, but maybe no need to bump all minor/major changes for all of them at the same time.

File: importer/src/cli.ts
Comment (SferaDev) [inline importer/src/cli.ts:17]:
```suggestion
  .option('--branch <branch>', 'The branch to import data to.', 'main')
```

File: importer/src/cli.ts
Comment (SferaDev) [inline importer/src/cli.ts:15]: Not sure I get the ``--create`` flag. If I don't pass it, it errors me out. If I do pass it still asks me for confirmation.

In case you don't pass it I would ask interactively and have the flag to not ask.

Screenshot 2022-05-04 at 08 20 19

File: importer/src/cli.ts
Comment (SferaDev) [inline importer/src/cli.ts:13]: I would have expected with this flag to use the existing table columns, but I guess order cannot be guaranteed. Could we make it interactive somehow? [just a comment, but I understand this as a future improvement]

**Review by SferaDev — COMMENTED** [review]
I would like to add new rows to an already existing table/schema. However it's not properly comparing the types from the online schema.

Screenshot 2022-05-04 at 08 44 48

Screenshot 2022-05-04 at 08 45 03

Screenshot 2022-05-04 at 08 29 07

**Comment by SferaDev** [comment]
Since we rely on the client we also have the createMany error handling issue

Screenshot 2022-05-04 at 08 31 36

giphy (1)

File: importer/package.json
Comment (SferaDev) [inline importer/package.json:30]: Not 100% sure about this dependency, seems not actively maintained

Screenshot 2022-05-04 at 08 37 34

File: importer/src/cli.ts
Comment (SferaDev) [inline importer/src/cli.ts:9]: I would add another option for ``delimiter`` and probably attempt to make it "smart".

Although CSV stands for comma-separated-values, the delimiter could be anything. Most popular are semicolons, tabs, spaces, etc...

File: importer/src/csv.ts
Comment (SferaDev) [inline importer/src/csv.ts:11]: Do we handle different encodings? I wouldn't assume UTF-8 by default, and maybe make it configurable

**Review by SferaDev — COMMENTED** [review]
@gimenete Works really good, I love the type inference 💯 

Added a lot of comments, most of them can be taken care after merging. I'll leave you decide how you want to prioritise them. Ping me if you need an approval.

File: importer/src/cli.ts
Comment (gimenete) [reply importer/src/cli.ts:9]: Good idea.

Question about one use case for that. How could we specify tab delimiters with flags in the CLI? 🤔

File: importer/src/csv.ts
Comment (gimenete) [reply importer/src/csv.ts:11]: Good suggestion. I'm not sure if the csv parser already takes care of that somehow (checking if there's a BOM for UTF), but I guess for most cases if the encoding is a different one, we'd need a way to specify it.

**Comment by gimenete** [comment]
> Since we rely on the client we also have the createMany error handling issue

Yup. I'll upgrade it once we release a new version of the client.

> I would like to add new rows to an already existing table/schema. However it's not properly comparing the types from the online schema. 

I'll look into that.

Thanks for your review!! 🙏

File: importer/src/cli.ts
Comment (SferaDev) [reply importer/src/cli.ts:9]: What about "\t"

File: importer/src/cli.ts
Comment (SferaDev) [reply importer/src/cli.ts:15]: Yeah, although I would make it always ask and maybe have a interactive/non-interactive flag that disables inquirer everywhere and fails if the flags for ``create`` or ``force`` are not provided. WDYT?

**Comment by gimenete** [comment]
@SferaDev the problem with the email/string thing should be fixed. I was not implementing the easiest case for `castType`: when the types are equal 🤦‍♂️ 

I've addressed some feedback and I think the rest can be implemented or explored in other PRs.

**Review by SferaDev — APPROVED** [review]
Tested again, now it works properly.

I would create several follow-up issues:

- Add delimiter option (semicolons, tabs, spaces)
- Review different encodings (latin1)
- Add a non-interactive flag, make all commands to inquiry instead of erroring out, if flags are provided ignore the inquiry question (related to https://github.com/xataio/client-ts/pull/127/files#r864487665)
- Add a way to call interactively as an npm package (API)

**Comment by gimenete** [comment]
Thanks @SferaDev for the review and the feedback! I've created https://github.com/xataio/client-ts/issues/142

**Pull request merged**
